pkg/lib/authn/identity/loginid/provider.go

```diff
@@ -85,6 +85,38 @@ func (p *Provider) GetByValue(value string) ([]*identity.LoginID, error) {
 	return is, nil
 }
 
+func (p *Provider) GetByKeyAndValue(key string, value string) (*identity.LoginID, error) {
+	// Normalize expects loginID is in correct type so we have to validate it first.
+	invalid := p.Checker.ValidateOne(identity.LoginIDSpec{
```

Comment:
No need to call ValidateOne. What you need to do is

1. Check `key` is a valid login ID key. if not, return `ErrGetUsersInvalidArgument.New("invalid Login ID key")` You do this by calling `LoginIDConfig.GetKeyConfig`
2. Normalize `value` according to the type.

Reply:
updated to use `LoginIDConfig`